model/src/main/kotlin/Scope.kt

```diff
@@ -47,6 +47,10 @@ data class Scope(
          */
         val dependencies: SortedSet<PackageReference>
 ) : Comparable<Scope> {
+    fun collectAllDependencies(): SortedSet<Identifier> = dependencies.map { it.id }.toSortedSet().also { result ->
```

Comment:
Here the return type can be omitted as it's not recursive like in `PackageReference`.

Reply:
Works, but I would like to keep it for consistency with Project (see below) and PackageReference.